src/string_bytes.cc

```diff
@@ -162,11 +173,11 @@ size_t hex_decode(char* buf,
                   const size_t srcLen) {
   size_t i;
   for (i = 0; i < len && i * 2 + 1 < srcLen; ++i) {
-    unsigned a = hex2bin(src[i * 2 + 0]);
-    unsigned b = hex2bin(src[i * 2 + 1]);
+    unsigned a = unhex(src[i * 2 + 0]);
+    unsigned b = unhex(src[i * 2 + 1]);
     if (!~a || !~b)
```

Comment:
Does this still work with `unhex()`’s return type changed? It seems to me some of the speedup may come from the compiler eliminating this branch completely because it knows the upper bits of `a` and `b` will never be set:

```
$ ./node-master -p 'Buffer(10).write("abcdxx", "hex")'
2
$ ./node -p 'Buffer(10).write("abcdxx", "hex")'
3
```

(aka we need more tests to cover these cases?)